src/main/scala/com/yotpo/metorikku/utils/FileUtils.scala

```diff
@@ -48,17 +48,25 @@ object FileUtils {
   }
 
   def readConfigurationFile(path: String): String = {
-    val envAndSystemProperties = System.getProperties().asScala ++= System.getenv().asScala
-    val prefix = envAndSystemProperties.get("CONFIG_FILES_PATH_PREFIX") match {
-      case Some(prefix) => prefix
-      case _ => ""
-    }
+    val envAndSystemProperties = getEnvProperties()
+    val prefix = getFilesPathPrefix(Option.apply(envAndSystemProperties)).getOrElse("")
 
     val fileContents = readFileWithHadoop(prefix + path)
     StringSubstitutor.replace(fileContents, envAndSystemProperties.asJava)
   }
 
+  def getEnvProperties(): Map[String, String] =  {
+    val envAndSystemProperties = System.getProperties().asScala ++= System.getenv().asScala
+    envAndSystemProperties.toMap
+  }
 
+  def getFilesPathPrefix(envProperties: Option[Map[String,String]]): Option[String] = {
+    envProperties.getOrElse(getEnvProperties()).get("CONFIG_FILES_PATH_PREFIX")
+  }
+
+  def getFailedDFPathPrefix(envProperties: Option[Map[String,String]]): Option[String] = {
```

Comment:
I don't really understand the generalization of getFilesPathPrefix and getFailedDFPathPrefix.
They are not used in the same scenario IMO.
Can you maybe explain?

Reply:
They both extract environment variables and use them in the main logic. Maybe it should be part of the `Configuration` class?

Reply:
discussed offline